src/App.tsx

```diff
@@ -1,63 +1,112 @@
-import React from 'react';
+import React, { useState } from 'react';
+import cn from 'classnames';
 import 'bulma/css/bulma.css';
 import './App.scss';
+import { GoodList } from './Components/GoodList/GoodList';
+import { goodsFromServer } from './Components/GoodsFromServer';
+import { SortType } from './types/SortType';
 
-export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType,
+  isReversed: boolean,
+) {
+  const preparedGoods: string[] = [...goods];
+
+  if (sortField) {
+    let sortFunction;
+
+    switch (sortField) {
+      case SortType.Name:
+        sortFunction = (
+          good1: string,
+          good2: string,
+        ) => good1.localeCompare(good2);
+        break;
+
+      case SortType.Length:
+        sortFunction = (
+          good1: string,
+          good2: string,
+        ) => good1.length - good2.length;
+        break;
+
+      default:
+        throw new Error(`Unknown sort field: ${sortField}`);
+    }
+
+    if (sortFunction) {
+      preparedGoods.sort(sortFunction);
+    }
```

Comment:
I agree it was my first solution
But now It's kinda upgraded, because this makes it possible to add or change sorting methods without having to change the code directly in the case blocks.